clientv3/balancer.go

```diff
@@ -255,8 +330,12 @@ func (b *simpleBalancer) Up(addr grpc.Address) func(error) {
 	b.readyOnce.Do(func() { close(b.readyc) })
 	return func(err error) {
 		b.mu.Lock()
+		if err.Error() == "grpc: failed with network I/O error" ||
+			err.Error() == "grpc: the connection is drained" {
+			setState(b.addrs, addr.Addr, addrStateTransientFailure)
```

Comment:
`setState` is only used for addrStateTransientFailure; it would be more straightforward to get rid of the state entirely:

```go
type addrConn struct {
    addr       grpc.Address
    lastFailed time.Time
}
...
b.addrs.fail(addr.Addr)
```